nipype/interfaces/workbench/metric.py

```diff
@@ -0,0 +1,147 @@
+        xor=["area_metrics"],
+        desc="Specify surfaces to do vertex area correction based on")
+    area_metrics = traits.Bool(
+        position=5,
+        argstr="-area-metrics",
+        xor=["area_surfs"],
+        desc="Specify vertex area metrics to do area correction based on")
+    current_area = File(
+        exists=True,
+        position=6,
+        argstr="%s",
+        desc="A relevant anatomical surface with <current-sphere> mesh OR"
+             " a metric file with vertex areas for <current-sphere> mesh")
+    new_area = File(
+        exists=True,
+        position=7,
+        argstr="%s",
+        desc="A relevant anatomical surface with <current-sphere> mesh OR"
+             " a metric file with vertex areas for <current-sphere> mesh")
+    roi_metric = File(
+        exists=True,
+        position=8,
+        argstr="-current-roi %s",
+        desc="Input roi on the current mesh used to exclude non-data vertices")
+    valid_roi_out = traits.Bool(
+        position=9,
+        argstr="-valid-roi-out",
+        desc="Output the ROI of vertices that got data from valid source vertices")
+    largest = traits.Bool(
+        position=10,
+        argstr="-largest",
+        desc="Use only the value of the vertex with the largest weight")
+
+
+class MetricResampleOutputSpec(TraitedSpec):
+    out_file = File(exists=True, desc="the output metric")
+    roi_file = File(desc="ROI of vertices that got data from valid source vertices")
+
+
+class MetricResample(WBCommand):
+    """
+    Resample a metric file to a different mesh
+
+    >>> from nipype.interfaces.workbench import MetricResample
+    >>> metres = MetricResample()
+    >>> metres.inputs.in_file = 'sub-01_task-rest_bold_space-fsaverage5.L.func.gii'
+    >>> metres.inputs.method = 'ADAP_BARY_AREA'
+    >>> metres.inputs.current_sphere = 'fsaverage5_std_sphere.L.10k_fsavg_L.surf.gii'
+    >>> metres.inputs.new_sphere = 'fs_LR-deformed_to-fsaverage.L.sphere.32k_fs_LR.surf.gii'
+    >>> metres.inputs.area_metrics = True
+    >>> metres.inputs.current_area = 'fsaverage5.L.midthickness_va_avg.10k_fsavg_L.shape.gii'
+    >>> metres.inputs.new_area = 'fs_LR.L.midthickness_va_avg.32k_fs_LR.shape.gii'
+    >>> metres.cmdline
+    'wb_command -metric-resample sub-01_task-rest_bold_space-fsaverage5.L.func.gii \
+    fsaverage5_std_sphere.L.10k_fsavg_L.surf.gii \
+    fs_LR-deformed_to-fsaverage.L.sphere.32k_fs_LR.surf.gii \
+    ADAP_BARY_AREA fs_LR-deformed_to-fsaverage.L.sphere.32k_fs_LR.surf.out \
+    -area-metrics fsaverage5.L.midthickness_va_avg.10k_fsavg_L.shape.gii \
+    fs_LR.L.midthickness_va_avg.32k_fs_LR.shape.gii'
+    """
+    input_spec = MetricResampleInputSpec
+    output_spec = MetricResampleOutputSpec
+    _cmd = 'wb_command -metric-resample'
+
+    def _format_arg(self, opt, spec, val):
+        if opt == "out_file":
+            # ensure generated filename is assigned to trait
+            self.inputs.trait_set(out_file=val)
```

Comment:
Sorry, I meant to say that I think this (and the `_list_outputs` entry) shouldn't be necessary if you make the input spec match the output spec.